src/options.js

```diff
@@ -0,0 +1,215 @@
+                        if (lastError) {
+                            console.log(lastError.message);
+                            $("#new_auth_url_msg").text(lastError.message);
+                        }
+                    }
+                });
+            }
+        });
+    }
+}
+
+function removeAuthorizedSite(id, URL) {
+    //console.debug(id);
+    //console.debug(URL);
+    var r = confirm("Really remove/disable " + URL + "?");
+    if (r) {
+        chrome.permissions.contains({
+            origins: [URL]
+        }, function(result) {
+            if (result) {
+                // The extension has the permissions.
+                if (URL == "https://gist.github.com/*") {
+                    chrome.storage.sync.set({ public_gist_enabled: false });
+                    $("#new_auth_url_" + id)
+                        .find("input")
+                        .removeClass("btn-danger")
+                        .addClass("btn-primary")
+                        .val("Enable")
+                        .unbind("click")
+                        .bind('click', function() {
+                            addAuthorizedSite(id, URL);
+                        });
+                } else if (URL == "https://github.com/*") {
+                    chrome.storage.sync.set({ public_github_enabled: false });
+                    $("#new_auth_url_" + id)
+                        .find("input")
+                        .removeClass("btn-danger")
+                        .addClass("btn-primary")
+                        .val("Enable")
+                        .unbind("click")
+                        .bind('click', function() {
+                            addAuthorizedSite(id, URL);
+                        });
+                } else {
+                    chrome.permissions.remove({
+                        origins: [URL]
+                        }, function(removed) {
+                        if (removed) {
+                            // The permissions have been removed.
+                            $("#new_auth_url_" + id).remove();
+                        } else {
+                            // The permissions have not been removed (e.g., you tried to remove
+                            // required permissions).
+                            alert("Required permissions cannot be removed.")
+                        }
+                    });
+                }
+            } else {
+              // The extension doesn't have the permissions.
+              console.debug("not found");
+            }
+        });
+    }
+}
+
+function setupUI() {
+    showAuthorizedSites();
+    $("#new_auth_url_submit").click(addAuthorizedSite);
+    $("#new_auth_url").keyup( function () {
+        if($(this).val() == '' ) {
+            $("#new_auth_url_submit").prop("disabled", true);
+        } else {
+            $("#new_auth_url_submit").prop("disabled", false);
+        }
+    });
+}
+
+document.addEventListener('DOMContentLoaded', setupUI);
```

Comment:
Nit: Line break at the end of file. https://robots.thoughtbot.com/no-newline-at-end-of-file